src/api/v4/dataview/category.js

```diff
@@ -0,0 +1,240 @@
+var _ = require('underscore');
+var Base = require('./base');
+var constants = require('../constants');
+var CategoryDataviewModel = require('../../../dataviews/category-dataview-model');
+var CategoryFilter = require('../../../windshaft/filters/category');
+
+/**
+ * Category dataview object
+ *
+ * @param {carto.source.Base} source - The source where the dataview will fetch the data.
+ * @param {string} column - The column name to get the data.
+ * @param {object} options
+ * @param {carto.operation} options.operation - The operation to apply to the data.
+ * @param {string} options.operationColumn - The column name used in the operation.
+ *
+ * @constructor
+ * @extends carto.dataview.Base
+ * @memberof carto.dataview
+ * @api
+ */
+function Category (source, column, options) {
+  this._initialize(source, column, options);
+}
+
+Category.prototype = Object.create(Base.prototype);
+
+/**
+ * Set the dataview maxCategories
+ *
+ * @param  {string} maxCategories
+ * @return {carto.dataview.Category} this
+ * @api
+ */
+Category.prototype.setMaxCategories = function (maxCategories) {
+  this._checkMaxCategories(maxCategories);
+  this._options.maxCategories = maxCategories;
+  if (this._internalModel) {
+    this._internalModel.set('max_categories', maxCategories);
+  }
+  return this;
+};
+
+/**
+ * Return the current dataview maxCategories
+ *
+ * @return {string} Current dataview maxCategories
+ * @api
+ */
+Category.prototype.getMaxCategories = function () {
+  return this._options.maxCategories;
+};
+
+/**
+ * Set the dataview operation
+ *
+ * @param  {carto.operation} operation
+ * @return {carto.dataview.Category} this
+ * @api
+ */
+Category.prototype.setOperation = function (operation) {
+  this._checkOperation(operation);
+  this._options.operation = operation;
```

Comment:
We store "options" as private instance variables in layers, sources, and styles. We should unify both approaches! Why did you go with `this._options`?

Reply:
I think it's because we set the variable `this._options` in Base. But we can then assign those options to variable instances.

So, OK, let's go with separated variable instances.